cmd/tealdbg/debugger.go

```diff
@@ -364,28 +364,26 @@ func (s *session) GetSourceMap() ([]byte, error) {
 	lines := make([]string, len(s.lines))
 	const targetCol int = 0
 	const sourceIdx int = 0
-	sourceLine := 0
-	const sourceCol int = 0
 	prevSourceLine := 0
 
 	// the very first entry is needed by CDT
-	lines[0] = logic.MakeSourceMapLine(targetCol, sourceIdx, 0, sourceCol)
+	lines[0] = logic.MakeSourceMapLine(targetCol, sourceIdx, 0, 0)
 	for targetLine := 1; targetLine < len(s.lines); targetLine++ {
 		if pc, ok := s.pcOffset[targetLine]; ok && pc != 0 {
-			sourceLine, ok = s.offsetToLine[pc]
+			source, ok := s.offsetToSource[pc]
 			if !ok {
 				lines[targetLine] = ""
 			} else {
-				lines[targetLine] = logic.MakeSourceMapLine(targetCol, sourceIdx, sourceLine-prevSourceLine, sourceCol)
-				prevSourceLine = sourceLine
+				lines[targetLine] = logic.MakeSourceMapLine(targetCol, sourceIdx, source.Line-prevSourceLine, source.Column)
+				prevSourceLine = source.Line
 			}
 		} else {
 			delta := 0
 			// the very last empty line, increment by number src number by 1
 			if targetLine == len(s.lines)-1 {
 				delta = 1
 			}
-			lines[targetLine] = logic.MakeSourceMapLine(targetCol, sourceIdx, delta, sourceCol)
+			lines[targetLine] = logic.MakeSourceMapLine(targetCol, sourceIdx, delta, 0)
```

Comment:
Assuming you take the previous suggestion, this might work better, since it'll reset the column index to 0.
```suggestion
			lines[targetLine] = logic.MakeSourceMapLine(targetCol, sourceIdx, delta, -prevLocation.Column)
```

Reply:
I think this case happens when two or more consecutive lines in the disassembly map to the same PC. For example:
```
label:
   pop
```
Both point to the same pc.  Therefore only the first line is in `s.pcOffset`, so `ok` is false when doing the second line.

I think the source line and col for the second line should be exactly the same as for the first, so I think we should stick with a 0 here (recognizing that it's a delta).

Perhaps we should use `-prevLocation.Column` for the "very last empty line" case though.

I was confused by this for a while, and can certainly see why you wanted to skip it.  If this makes sense to you, let's do it. Otherwise, I will change this PR to pass the new `SourceLocation` map, but keep the code exactly as it was, except for using `s.offsetToSource[pc].Line` to get the line. (All columns will stay 0.)

Reply:
Good point, I misunderstood what was happening here. It appears this is actually creating a source map from a disassembled TEAL program to another TEAL program, unlike `logic.GetSourceMap`, which creates a map from a compiled program to the input TEAL program.

That being said, I believe your suggestions are all correct, so I will approve after you make that change.